### PR TITLE
1.10.0: Notifications & Comments improvements

### PR DESCRIPTION
**Pull request by GuillaumeSalles**

### CODE REVIEW

**Comment by colindjk** [comment]
This PR needs to change the way you interact with `useSyncExternalStoreWithSelector`.

If you instantiate the selector on every render, the returned object will be a new reference. Reference [here](https://github.com/facebook/react/blob/9fcaf88d58cfd942e2fdd303ae8291dbf4828969/packages/use-sync-external-store/src/useSyncExternalStoreWithSelector.js#L102).

Lines to fix (Only for `packages/liveblocks-react/src/liveblocks.tsx`). 
- [useInboxNotifications](https://github.com/liveblocks/liveblocks/blob/47dbff90b612a882d0782bafd6d994f8271b5a1b/packages/liveblocks-react/src/liveblocks.tsx#L159)
- [useInboxNotificationsSuspense](https://github.com/liveblocks/liveblocks/blob/47dbff90b612a882d0782bafd6d994f8271b5a1b/packages/liveblocks-react/src/liveblocks.tsx#L205)
- [useUnreadInboxNotificationsCount](https://github.com/liveblocks/liveblocks/blob/47dbff90b612a882d0782bafd6d994f8271b5a1b/packages/liveblocks-react/src/liveblocks.tsx#L239)
- [useUnreadInboxNotificationsCountSuspense](https://github.com/liveblocks/liveblocks/blob/47dbff90b612a882d0782bafd6d994f8271b5a1b/packages/liveblocks-react/src/liveblocks.tsx#L287)
- [useThreadFromCache](https://github.com/liveblocks/liveblocks/blob/47dbff90b612a882d0782bafd6d994f8271b5a1b/packages/liveblocks-react/src/liveblocks.tsx#L382)

File: packages/liveblocks-core/src/room.ts [hunk 1]
```diff
@@ -456,25 +488,121 @@ type SubscribeFn<
...
+    emoji: string;
+  }): Promise<void>;
+};
+
 export type Room<
   TPresence extends JsonObject,
   TStorage extends LsonObject,
   TUserMeta extends BaseUserMeta,
   TRoomEvent extends Json,
 > = CommentsApi<any /* TODO: Remove this any by adding a proper thread metadata on the Room type */> & {
```
Comment (colindjk) [inline packages/liveblocks-core/src/room.ts:591]: Can be a fix in here for the `any` in `CommentsApi`? 

And/or, expose the `CommentsApi` type from the root folder to let users of the liveblocks sdk handle this themselves? Would help with extending that interface for custom hooks etc.

File: docs/pages/products/comments/notification-inbox.mdx [hunk 2]
```diff
@@ -2,7 +2,130 @@
 meta:
   title: "Notification inbox"
   parentTitle: "Comments"
-  description: "..."
+  description: "Show a list of notifications"
 ---
 
-Coming soon...
+TODO: Introduction
```
Comment (FlowFlorent) [inline docs/pages/products/comments/notification-inbox.mdx:8]: Reminder to not forget

**Review by marcbouchenoire — APPROVED** [review]
🔔🔔🔔

**Pull request merged**